### DISCUSSION TITLE
Does the marginal coverage guarantee hold for Tweedie on skewed tails?

### DISCUSSION THREAD

**Discussion by burning-cost** [burning-cost/insurance-conformal#3] [Q&A]
The coverage guarantee for conformal prediction is usually stated as marginal — 1-alpha coverage over the joint distribution of (X, Y). My concern is that for insurance Tweedie data, the tails are very heavy and sparse. A guarantee that holds marginally might give terrible conditional coverage in the tail, which is exactly where pricing errors are most costly.

Specifically: if I set alpha=0.1 and my Tweedie p parameter is around 1.6, should I expect 90% coverage to hold for large losses (say, above the 95th percentile)? Or does the coverage collapse in the tail?

We have about 80k policies in the calibration set.

**Comment by burning-cost** [comment]
You have identified the core tension correctly. The marginal coverage guarantee is real and finite-sample valid (with the usual +1/n correction), but conditional coverage — coverage conditional on X being in some region — is a much harder property and standard split conformal does not guarantee it.

For Tweedie with p ~= 1.6 on UK motor data, what typically happens:

- **Marginal coverage holds well** at 80k calibration records. The finite-sample guarantee is exact up to the +1/(n+1) correction, so your empirical coverage will be within ~0.001 of the nominal level overall.
- **Conditional coverage in the tail is worse than nominal.** This is structural, not a bug. In the 95th–99th percentile loss range, you will typically see conditional coverage of 70–85% rather than 90%, because the nonconformity scores in the tail have a different distribution to the bulk.

**What to do about it:**

**Option 1: Use a tail-stratified calibration.** Split your calibration set into two strata — below and above the 90th percentile of predicted premium — and fit separate quantile corrections for each stratum. This is rough but effective:

```python
from insurance_conformal import TweedieConformalPredictor, StratifiedCalibrator

model = TweedieConformalPredictor(
    base_model=your_catboost_model,
    calibrator=StratifiedCalibrator(
        split_col="predicted_premium",
        split_quantile=0.90,
        alpha=0.10
    )
)
model.calibrate(cal_df)
intervals = model.predict_intervals(test_df)
```

**Option 2: Use a conformalized quantile regression (CQR) score.** Instead of using absolute or normalised residuals as the nonconformity score, use the gap between the observed value and the predicted quantile bounds from a quantile CatBoost. CQR has better conditional coverage properties than standard split conformal on skewed data:

```python
from insurance_conformal import CQRTweediePredictor

# CQR fits quantile models at alpha/2 and 1-alpha/2 as the base
model = CQRTweediePredictor(alpha=0.10, quantile_model="catboost")
model.fit(train_df, outcome_col="loss_cost", feature_cols=features)
model.calibrate(cal_df)
intervals = model.predict_intervals(test_df)
```

CQR is the approach I would recommend for pricing use cases. The intervals are tighter in the body and better calibrated in the tail. The cost is that you need to fit two additional quantile models.

**Practical check:** always plot conditional coverage as a function of the percentile of predicted premium on a held-out test set before using intervals in production. A 5-point coverage shortfall in the top decile is usually acceptable; a 15-point shortfall is not.
